fe/fe-core/src/main/java/com/starrocks/catalog/ForeignKeyConstraint.java

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.starrocks.catalog;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Streams;
+import com.starrocks.common.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.spark_project.guava.base.Strings;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+// FOREIGN KEY: (column1, column2) REFERENCES catalog.database.table(column1', column2')
+// if catalog or database is not provided, use current catalog or current database instead.
+public class ForeignKeyConstraint {
+    private static final Logger LOG = LogManager.getLogger(ForeignKeyConstraint.class);
+
+    private static final String FOREIGN_KEY_REGEX = "\\(((,?\\s*\\w+\\s*)+)\\)\\s+((?i)REFERENCES)\\s+" +
```

Comment:
Add a TODO to use StarRocks's parser later, we can support constraints in our SQL directly later?